### ISSUE TITLE
OAI GetRecord request still giving dcterms specifics fields ?

### ISSUE THREAD

**Issue by rmo-cdsp** [IQSS/dataverse#3563]
Hello !

I think this issue is related to #3368 . We updated our dataverse with v4.6, hoping to use the reworked OAI-PMH harvesting protocol of Dataverse. The thing is, it seems we still get not oai-dc elements (as isReferencedBy or dateSubmitted, for example). I thought we failed our update, so I tried on dataverse.harvard.eu, just in case, but the same thing happened. Here is an example of request that "should" (I may have done a mistake on my side) work : 
https://dataverse.harvard.edu/oai?verb=ListRecords&metadataPrefix=oai_dc

As you can see, there are isReferencedBy, dateSubmitted or spatial fields, that are not part of oai-pmh fields. I tried to look at the code of v4.6 on github, and I can't figure out how, for now, this is happening. It's like if it was still like v5.1. Is the .war provided not including the OAI reworked version ? I'm wondering.

**Comment by djbrooke** [comment]
Hey @rmo-cdsp - thanks for creating a ticket for this based on our IRC conversation yesterday! I just talked about this with @landreev and he's going to see if there's anything unexpected here. If we need to make code changes, we'll discuss further. 

Tagging @jggautier as well because we also talked about this a bit yesterday.

**Comment by landreev** [comment]
Oh, thank you so much for pointing this out! 
The code that generates oai_dc in 4.6 has been updated. But you (and we!) are getting the old-style dc records via OAI *because they are cached*! The dataverse app knows to re-export records when the dataset has been changed... but it doesn't automatically re-export when the export code itself changes. So all these cached oai_dc records needed to be re-exported in a forced way; it was something we should have included as an upgrade step, in the 4.6 release notes.  But apparently forgot about.

For example, here's our oai_dc record: 

https://dataverse.harvard.edu/oai?verb=GetRecord&identifier=hdl:1902.1/18768&metadataPrefix=oai_dc

<OAI-PMH xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd">
<responseDate>2017-01-11T16:16:46Z</responseDate>
...
<dc:creator>S. Gaye</dc:creator><dc:publisher>Harvard Dataverse</dc:publisher><dc:issued>2012-09-24</dc:issued>
<dc:modified>2012-09-20T17:03:24Z</dc:modified>
<dc:description> ...

but then if I delete the cached copy: 

cd <MY FILES DIRECTORY>/1902.1/18768

ls -l  export_oai_dc.cached 
-rw-r--r-- 1 root root 5779 Aug 25 11:48 export_oai_dc.cached

rm -f export_oai_dc.cached

 - then go back to https://dataverse.harvard.edu/oai?verb=GetRecord&identifier=hdl:1902.1/18768&metadataPrefix=oai_dc: 

<OAI-PMH xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd">
<responseDate>2017-01-11T16:19:43Z</responseDate>
...
<dc:creator>S. Gaye</dc:creator>
<dc:publisher>Harvard Dataverse</dc:publisher>
<dc:description> ...

and, looking at the file system again: 

ls -l export_oai_dc.cached 
-rw-r--r-- 1 root root 1854 Jan 11 11:19 export_oai_dc.cached


So the easiest way to go about this is to wipe clean all the oai_dc caches. By something like 

cd <YOUR FILES DIRECTORY>
/bin/rm `find . -name export_oai_dc.cached`

(please be extra careful; make sure your files are backed up, etc.)

You could also do this by forcing a full reexport of all your datasets through the API; but that would also make your Dataverse re-export all the other formats, which have not changed. 

Once again, thank you for bringing this to our attention.

**Comment by landreev** [comment]
PLEASE NOTE that there were some things in angle brackets and backquotes in my last github comment, that github removed!
It should have been: 

cd &lt;MY FILES DIRECTORY&gt;/1902.1/18768

ls -l export_oai_dc.cached

... 

and 

cd &lt;YOUR FILES DIRECTORY&gt;

/bin/rm \'find . -name export_oai_dc.cached\'

**Comment by landreev** [comment]
Hmm, now it's showing single quotes around the "find ..." command. These should be backquotes/backticks! 
let me try again, using `&#96;`:

/bin/rm &#96;find . -name export_oai_dc.cached&#96;

**Comment by landreev** [comment]
(I'm wiping these old caches on our production server now)

**Comment by rmo-cdsp** [comment]
Hey ! Our sysadmin followed your instructions and we now have the v4.6 oai version, that's perfect ! Thank you for this.

**Comment by djbrooke** [comment]
Great news @rmo-cdsp and thanks @landreev !

**Comment by djbrooke** [comment]
@landreev is there some documentation update we need to make or can this issue be closed?

**Comment by rmo-cdsp** [comment]
Hello !

Our dataverse, in v4.6.2 (4.7.1 coming soon for us), had troubles to "update" its cached files, again. I thought the issue #3809 had a bug, given that no urls could be seen in the oai_pmh output, but it wasn't the case. It may be a problem for people not knowing this issue, because you have to know this exact problem. Is there anything intended to change the mecanism ? For example, while running a new export in the "Manage Harvesting Server", delete the cached files and make a new proper file ?

**Comment by pdurbin** [comment]
@rmo-cdsp no, we don't have any plans to change the code with regard to cached files. If you'd like the behavior to change, please open a new GitHub issue describing how you'd like it to work. Thanks!

**Comment by jggautier** [comment]
I'm going to close this issue, since I think it affected only older Dataverse repositories that needed to re-cache the oai_dc metadata they were publishing.

I'm thinking that @rmo-cdsp's suggestion ("while running a new export in the "Manage Harvesting Server", delete the cached files and make a new proper file") could go in another issue. (As of this post, it doesn't look like an issue for this has been created for this.)
